src/Tranche4626.sol

```diff
@@ -0,0 +1,152 @@
+// create mint flow
+// create redeem flow
+// create withdraw flow 
+// messages collectRedeem & collectInvest
+
+
+import "./token/restricted.sol";
+
+interface ConnectorLike {
+    function deposit(address _tranche, address _receiver, uint256 _assets) external returns (uint256);
+    function mint(address _tranche, address _receiver, uint256 _shares) external returns (uint256);
+    function maxDeposit(address _user, address _tranche) external returns (uint256);
+    function maxMint(address _user, address _tranche) external returns (uint256);
+    function maxWithdraw(address _user, address _tranche) external returns (uint256);
+    function maxRedeem(address _user, address _tranche) external returns (uint256) 
+}
+
+/// @title Tranche4626
+/// @author ilinzweilin
+contract Tranche4626 is RestrictedToken {
+
+    ConnectorLike public connector;
+
+    address public asset; // underlying stable ERC-20 stable currency.
+    uint256 public maxAssetDeposit = 2 ** 256 - 1; // max stable currency deposit into the tranche -> default: no limit.
+
+    uint128 latestPrice; // lates share / token price 
+    uint256 lastPriceUpdate; // timestamp of the latest share / token price update
+   
+    // ERC4626 events
+    event Deposit(address indexed sender, address indexed owner, uint256 assets, uint256 shares);
+    event Withdraw(address indexed sender, address indexed receiver, address indexed owner, uint256 assets, uint256 shares);
+    
+    constructor(address _asset, address _connector, uint8 _decimals, string memory _name, string memory _symbol) super(decimals) {
+        name = _name;
+        symbol = _symbol;
+        asset = _asset;
+        connector = ConnectorLike(_connector);
+
+        wards[_connector] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /// @dev The total amount of vault shares.
+    /// @return Total amount of the underlying vault assets including accrued interest.
+    function totalAssets() public view returns (uint256) {
+       return totalSupply() * latestPrice; 
+    }
+
+    /// @dev Calculates the amount of shares / tranche tokens that any user would get for the amount of assets provided. The calcultion is based on the token price from the most recent epoch retrieved from Centrifuge chain.
+    function convertToShares(uint256 _assets) public view returns (uint256 shares) {
+        // TODO: it should round DOWN if it’s calculating the amount of shares to issue to a user, given an amount of assets provided.
+        shares = _assets / latestPrice;
+    }
+
+    /// @dev Calculates the asset value for an amount of shares / tranche tokens provided. The calcultion is based on the token price from the most recent epoch retrieved from Centrifuge chain.
+    function convertToAssets(uint256 _shares) public view returns (uint256 assets) {
+        // TODO: it should round DOWN if it’s calculating the amount of assets to issue to a user, given an amount of shares provided.
+        assets = _shares * latestPrice;
```

Comment:
This is implicitly assuming that the underlying asset/stable currency is always 1$, no? I.e, imagine that a share's latest price is 3$ but the underlying pool currency is worth 0.8$, then this conversion won't be correct, right?